enn_zoo/enn_zoo/microrts/__init__.py

```diff
@@ -0,0 +1,360 @@
+from dataclasses import dataclass
+from tokenize import String
+from typing import Any, Dict, List, Mapping, MutableMapping, Optional, Sequence, Tuple
+import random
+from entity_gym.environment.environment import EntityObs
+import numpy as np
+import numpy.typing as npt
+from copy import deepcopy
+import os
+import gym_microrts
+from gym_microrts import microrts_ai
+import xml.etree.ElementTree as ET
+import json
+
+from entity_gym.environment import (
+    CategoricalAction,
+    CategoricalActionMask,
+    Entity,
+    Environment,
+    EpisodeStats,
+    ObsSpace,
+    CategoricalActionSpace,
+    ActionSpace,
+    Observation,
+    Action,
+    VecEnv,
+)
+
+import jpype
+from jpype.imports import registerDomain
+import jpype.imports
+from jpype.types import JArray, JInt
+
+
+class GymMicrorts(Environment):
+    """
+    Turn-based version of Snake with multiple snakes.
+    Each snake has a different color.
+    For each snake, Food of that color is placed randomly on the board.
+    Snakes can only eat Food of their color.
+    When a snake eats Food of the same color, it grows by one unit.
+    When a snake grows and it's length was less than 11, the player receives a reward of 0.1 / num_snakes.
+    The game ends when a snake collides with another snake, runs into a wall, eats Food of another color, or all snakes reach a length of 11.
+    """
+
+    def __init__(
+        self,
+        map_path: str = "maps/10x10/basesTwoWorkers10x10.xml",
+        partial_obs: bool = False,
+        reward_weight: npt.NDArray[np.float32] = np.array(
```

Comment:
If it requires this to be an ndarray, it won't be possible to pass with json `kwargs`. Maybe make it a list instead? Or even 6 separate params?

Reply:
I used a list instead.